valhalla/sif/dynamiccost.h

```diff
@@ -44,7 +44,8 @@
         range(rapidjson::get<decltype(range.def)>(json, json_key,                                    \
                                                   costing_options->has_##option_name##_case()        \
                                                       ? costing_options->option_name()               \
-                                                      : range.def)));                                \
+                                                      : range.def),                                  \
+              json_key));                                                                            \
```

Comment:
then over here we check if that boolean was `true` meaning the range default clamped it, and if so we add a warning to whatever place we have to add warnings